mesheryctl/internal/cli/root/components/list.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2024 Meshery Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package components
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/eiannone/keyboard"
+	"github.com/fatih/color"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/server/models"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// represents the mesheryctl exp components list command
+var listComponentCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List registered components",
+	Long:  "List all components registered in Meshery Server",
+	Example: `
+	// View list of components
+mesheryctl exp components list
+// View list of components with specified page number (25 components per page)
+mesheryctl exp components list --page 2
+// View Total number of components
+mesheryctl exp components list --count
```

Comment:
Using mesheryctl exp components --count would enhance clarity and consistency compared to incorporating the count flag within mesheryctl exp components list --count.

Reply:
Done